discover/deployment/articles/03-configuring-liferay/03-liferay-portal-clustering/01-cluster-installation.markdown

```diff
@@ -0,0 +1,44 @@
+# Building Clustering for @product@ [](id=building-clustering)
+
+In order to use RMI-based clustering with @product@, a series of modules will need to be 
+compiled from source and several modules will need to be blacklisted to prevent them from
+starting up with the platform.
+
+## Build Liferay Portal from source
+
+These instructions assume building against master but will also work with the latest branch version.
+
+1.  Clone @product@ repo on Github:
+
+        git clone https://github.com/liferay/liferay-portal —branch master —single-branch --depth 1
+
+2.  Build master from source:
+
+        cd liferay-portal
+        ant all
+
+3.  Remove the following jars from `../bundles/osgi/portal`:
+
+        rm -f ../bundles/osgi/portal/com.liferay.portal.cache.single.jar
+        rm -f ../bundles/osgi/portal/com.liferay.portal.cluster.single.jar
+        rm -f ../bundles/osgi/portal/com.liferay.portal.scheduler.single.jar 
+
+## Build and Deploy Clustering Modules
+Build and deploy the following modules to `../bundles/osgi/portal`:
+
+        cd modules
+        ../gradlew :apps:foundation:portal:portal-cluster-multiple:deploy
+        ../gradlew :apps:foundation:portal-cache:portal-cache-ehcache-multiple:deploy
+        ../gradlew :apps:foundation:portal-scheduler:portal-scheduler-multiple:deploy
```

Comment:
@sammonsjl - What are the names of the JARs these commands build and deploy?

… I ask because I imagine admins will want to copy them into their existing portal installation. Thanks.

Reply:
Hey @jhinkey - they are:
```
com.liferay.portal.cache.ehcache.multiple.jar
com.liferay.portal.cluster.multiple.jar
com.liferay.portal.scheduler.multiple.jar
```
Also I just realized with the big source changes in 7.1, the above paths no longer contain "foundation" for example: 
```
./gradlew :apps:foundation:portal:portal-cluster-multiple:deploy
``` 
is now:
```
 ./gradlew :apps:portal:portal-cluster-multiple:deploy
```

Reply:
Gotcha. Thanks.